docs: Updated troubleshoot guide to show how to extract ErrorInfo from RpcException (#9484)
googleapis/google-cloud-dotnet: docs/devsite-help/troubleshooting.md

`Grpc.Core` from your application, which will ensure that the
native libraries are copied appropriately.

> ## How to get ErrorInfo from SpannerException

Reviewer: No, "How to get..." isn't a question - it's just information. (It would be "How can I get ErrorInfo from SpannerException?" or similar.)

But yes, I'd make this primarily about RpcException.

(This is my one drive-by comment...)

The ErrorInfo message provides error information that both humans and
applications can depend on. If SpannerException contains RPCException

> as it's inner execption, we can extract ErrorInfo from SpannerException.

Reviewer 2: ```suggestion
as its inner exception, we can extract ErrorInfo from SpannerException.
```

Reviewer: I'd suggest also using backticks for `ErrorInfo` and `SpannerException`

`Grpc.Core` from your application, which will ensure that the
native libraries are copied appropriately.

## How to get ErrorInfo from SpannerException

> The ErrorInfo message provides error information that both humans and

Reviewer: ```suggestion
The `ErrorInfo` message provides error information that both humans and
```

> applications can depend on. If SpannerException contains RPCException

Reviewer: ```suggestion
applications can depend on. If a `SpannerException` contains an `RpcException`
```

as it's inner execption, we can extract ErrorInfo from SpannerException.

The recommended way of doing this:

> ```csharp
> try
> {
>   // some spanner code
> }
> catch (SpannerException ex) when (ex.InnerException is RpcException rpcException )
>  {
>    var errorInfo = rpcException.GetErrorInfo();
> 
>    // Domain is the logical grouping to which the "reason" belongs.   
>    var domain = errorInfo.Domain;
> 
>    // The reason for the error. Error reasons are unique within a particular domain of errors.
>    var reason = errorInfo.Reason;
> 
>    // Additional structured details about this error.
>    var metadata = errorInfo.Metadata;
> 
>    // Log domain , reason and metadata of ErrorInfo.
>     }
> ```

Reviewer: ```suggestion
```csharp
try
{
    // Code using Spanner
}
catch (SpannerException ex) when (ex.InnerException is RpcException rpcException )
{
    var errorInfo = rpcException.GetErrorInfo();

    // Domain is the logical grouping to which the "reason" belongs.   
    var domain = errorInfo.Domain;

    // The reason for the error. Error reasons are unique within a particular domain of errors.
    var reason = errorInfo.Reason;

    // Additional structured details about this error.
    var metadata = errorInfo.Metadata;

    // Use the domain, reason and metadata of ErrorInfo, e.g. for logging or taking remedial action.
}
```
